misaligned.c

```diff
@@ -1,21 +1,52 @@
-#include <stdio.h>
+#include "misaligned.h"
 #include <assert.h>
 
-int printColorMap() {
-    const char* majorColor[] = {"White", "Red", "Black", "Yellow", "Violet"};
-    const char* minorColor[] = {"Blue", "Orange", "Green", "Brown", "Slate"};
-    int i = 0, j = 0;
-    for(i = 0; i < 5; i++) {
-        for(j = 0; j < 5; j++) {
-            printf("%d | %s | %s\n", i * 5 + j, majorColor[i], minorColor[i]);
+
+void referenceManual()
+{
+    int majorIndex = 0;
+    int minorIndex = 0;
+    int colorPairNumber = 0;
+    int colorPairLength = 0;
+    for(majorIndex = 0; majorIndex < numberOfMajorColors; majorIndex++) 
+    {
+        for(minorIndex = 1; minorIndex <= numberOfMinorColors; minorIndex++) 
+        {
+            colorPairLength = sprintf(&correctColorMap[colorPairNumber], "%d\t|%s\t|%s\n", majorIndex * numberOfMajorColors + minorIndex, majorColor[majorIndex], minorColor[minorIndex-1]);
+            colorPairNumber = colorPairNumber + colorPairLength;
+        }
+    }
+}
+
+int printColorMap() 
+{
+    int majorIndex = 0;
+    int minorIndex = 0;
+    int colorPairNumber = 0;
+    int colorPairLength = 0;
+    for(majorIndex = 0; majorIndex < numberOfMajorColors; majorIndex++) 
+    {
+        for(minorIndex = 0; minorIndex < numberOfMinorColors; minorIndex++) 
+        {
+            colorPairLength = sprintf(&colorMap[colorPairNumber], "%d |%s |%s\n", (majorIndex * numberOfMajorColors + minorIndex)+1, majorColor[majorIndex], minorColor[minorIndex]);
+            colorPairNumber = colorPairNumber + colorPairLength;
         }
     }
-    return i * j;
+    return majorIndex * minorIndex;
+}
+
+void testAlignment()
+{
+    int result;
+    referenceManual();
+    result = strcmp(&correctColorMap[0], &colorMap[0]);

```

Comment:
Since you already have the content that can be verified, please add test cases to check pair number where major color ends and another major color starts. for ex. 1, 5, 6, 10,11...